tensorflow/c/experimental/filesystem/filesystem_interface.h

```diff
@@ -84,22 +84,28 @@ typedef struct TF_TransactionToken {
 } TF_TransactionToken;
 
 typedef struct TF_Filesystem_Option_Value {
-  int type_tag;
-  int num_values;
+  int type_tag; // type of the value (int, real, buffer)
   union {
     int64_t inv_val;
     double real_val;
     struct {
       char* buf;
       int buf_length;
     } buffer_val;
-  } * values;  // owned
+  } u;
 } TF_Filesystem_Option_Value;
 
+typedef enum TF_Filesystem_Option_Type {
+  TF_Filesystem_Option_Type_Int = 0,
+  TF_Filesystem_Option_Type_Real,
+  TF_Filesystem_Option_Type_Buffer,
+} TF_Filesystem_Option_Type;
```

Comment:
```suggestion
  TF_Filesystem_Option_Type_Buffer,
  TF_Filesystem_Num_Option_Types, // must always be the last item
} TF_Filesystem_Option_Type;
```

This would allow code that checks that the options are valid for the current plugin version and makes the code be forward and backwards compatible.